### PR TITLE
Header Redesign / by hashicorp

### PR DESCRIPTION
**Pull request by captainill**
Added a bunch of files to easily share the styles across projects. I was also working on Terraform to make sure much of my shared code was easily transferrable to the other projects.

### CODE REVIEW

File: website/source/assets/javascripts/app/Sidebar.js [hunk 1]
```diff
@@ -12,11 +12,10 @@ Sidebar = Base.extend({
 	constructor: function(){
 		this.$body = $('body');
 		this.$overlay = $('.sidebar-overlay');
-    	this.$sidebar = $('#sidebar');
```
Comment (cbednarski) [inline website/source/assets/javascripts/app/Sidebar.js]: Some tabs vs. spaces thing going on here.

File: website/source/assets/stylesheets/hashicorp-shared/_hashicorp-header.scss [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                svg path{
+                    @include transition( all 300ms ease-in );
+                }
+            }
+
+            &.download{
+                margin-right: 10px;
+            }
+
+            >a {
```
Comment (cbednarski) [inline website/source/assets/stylesheets/hashicorp-shared/_hashicorp-header.scss]: Is there a semantic difference between `>a` and `> a`?

File: website/source/assets/javascripts/app/Sidebar.js (hunk 1 above)
Comment (captainill) [inline website/source/assets/javascripts/app/Sidebar.js]: thanks cleaned up all whitespace for the files.

File: website/source/assets/stylesheets/hashicorp-shared/_hashicorp-header.scss (hunk 2 above)
Comment (captainill) [inline website/source/assets/stylesheets/hashicorp-shared/_hashicorp-header.scss]: I don't believe there's any semantic diff. I updated >a to > a

**Comment by pearkes** [comment]
On retina the 2x logo still looks a little blurry. Is that just me or can you confirm @captainill?

**Comment by pearkes** [comment]
Otherwise this LGTM!

**Comment by mitchellh** [comment]
I'm keeping an eye on this, once we get the retina stuff fixed or cleared up, I'll merge, but we can't deploy the site for awhile so this can't deploy until 0.2.

**Comment by captainill** [comment]
@pearkes @mitchellh  thanks that should be fixed now if you wanna give another look.

**Comment by pearkes** [comment]
LGTM!

**Comment by mitchellh** [comment]
Let's do this.

**Pull request merged**